src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -994,14 +994,74 @@ public static ConnectionMultiplexer Connect(string configuration, TextWriter log
+        /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
+        public static ConnectionMultiplexer SentinelConnect(string configuration, TextWriter log = null)
+        {
+            var options = ConfigurationOptions.Parse(configuration);
+            return SentinelConnect(options);
+        }
+
+        /// <summary>
+        /// Create a new ConnectionMultiplexer instance that connects to a sentinel server
+        /// </summary>
+        /// <param name="configuration">The configuration options to use for this multiplexer.</param>
+        /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
+        public static ConnectionMultiplexer SentinelConnect(ConfigurationOptions configuration, TextWriter log = null)
+        {
+            if (string.IsNullOrEmpty(configuration.ServiceName))
+                throw new ArgumentException("A ServiceName must be specified.");
+
+            var sentinelConfigurationOptions = configuration.Clone();
+
+            // this is required when connecting to sentinel servers
+            sentinelConfigurationOptions.TieBreaker = "";
+            sentinelConfigurationOptions.CommandMap = CommandMap.Sentinel;
+
+            // use default sentinel port
+            sentinelConfigurationOptions.EndPoints.SetDefaultPorts(26379);
+
+            return Connect(sentinelConfigurationOptions, log);
+        }
+
+        /// <summary>
+        /// Create a new ConnectionMultiplexer instance that connects to a sentinel server, discovers the current master server
+        /// for the specified ServiceName in the config and returns a managed connection to the current master server
+        /// </summary>
+        /// <param name="configuration">The string configuration to use for this multiplexer.</param>
+        /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
+        public static ConnectionMultiplexer SentinelMasterConnect(string configuration, TextWriter log = null)
+        {
+            var options = ConfigurationOptions.Parse(configuration);
+            var sentinelConnection = SentinelConnect(options, log);
+
+            return sentinelConnection.GetSentinelMasterConnection(options, log);
+        }
+
+        /// <summary>
+        /// Create a new ConnectionMultiplexer instance that connects to a sentinel server, discovers the current master server
+        /// for the specified ServiceName in the config and returns a managed connection to the current master server
+        /// </summary>
+        /// <param name="configuration">The configuration options to use for this multiplexer.</param>
+        /// <param name="log">The <see cref="TextWriter"/> to log to.</param>
+        public static ConnectionMultiplexer SentinelMasterConnect(ConfigurationOptions configuration, TextWriter log = null)
+        {
+            var sentinelConnection = SentinelConnect(configuration, log);
```

Comment:
what is the lifetime of `sentinelConnection` here? it feels like the initial discovery connection could be `using` ?

Reply:
I believe this connection is kept open to be able to query the sentinel server for master changes.

Reply:
It can be GCed right after this method, so that doesn't seem right?

Reply:
Hmm... I know somewhere in the call stack it's setting up a subscription on the sentinel server to listen to events and I would think that would be holding onto a reference to that object, but I am not sure on that. I guess we need to figure out some way to test this and also to make it so that the 2 connections are linked and disposed together when the outer connection is disposed.

Reply:
I've updated this to keep a reference to the sentinel connection and to dispose it when the outer managed connection is disposed.

Reply:
Gotcha - I put some time to step through this with Marc live tomorrow - leaving other notes now!